test/database/dao/GuestDataDAOTest.java

```diff
@@ -1,22 +1,150 @@
+		GuestDataDTO guest2=new GuestDataDTO("Pablo","Salinas",LocalDateTime.of(1993,12,12,10,30),NationalityDTO.ARGENTIN,"1234569",bookingSaved2.getId());	
+		GuestDataDTO guest3=new GuestDataDTO("Sebastian","Cruz",LocalDateTime.of(1993,10,13,10,30),NationalityDTO.ARGENTIN,"1234568",bookingSaved3.getId());	
+	
+		GuestDataDTO guestSaved = guestDataDAO.save(guest);
+		GuestDataDTO guestSaved2 = guestDataDAO.save(guest2);
+		GuestDataDTO guestSaved3 = guestDataDAO.save(guest3);
+		
+		List<GuestDataDTO> searchGuestList = guestDataDAO.searchGuestList();
+		assertEqualsGuestDataDTO(guestSaved,searchGuestList.get(0));
+		assertEqualsGuestDataDTO(guestSaved2,searchGuestList.get(1));
+		assertEqualsGuestDataDTO(guestSaved3,searchGuestList.get(2));
+		
+		int deletedRecords = guestDataDAO.delete(guestSaved.getId());	
+		int deletedRecords2 = guestDataDAO.delete(guestSaved2.getId());
+		int deletedRecords3 = guestDataDAO.delete(guestSaved3.getId());
+		Assert.assertTrue(1 == deletedRecords);
+		Assert.assertTrue(1 == deletedRecords2);
+		Assert.assertTrue(1 == deletedRecords3);
+		
+		bookingDataDAO.delete(bookingSaved.getId());
+		bookingDataDAO.delete(bookingSaved2.getId());
+		bookingDataDAO.delete(bookingSaved3.getId());
+	}	
+	
+	@Test
+	public void testGetGuestAttributes() {
+		BookingDataDTO bookingDataDTO=new BookingDataDTO(LocalDateTime.now(),LocalDateTime.of(2023,12,12,10,30),PaymentMethodDTO.CASH,new BigDecimal("6000"));
+		BookingDataDTO bookingSaved = bookingDataDAO.save(bookingDataDTO);
+		
+		try {
+			GuestDataDTO guest=new GuestDataDTO(null,null,LocalDateTime.of(1990,12,12,10,30),NationalityDTO.ARGENTIN,"1234567",bookingSaved.getId());
+			
+			guestDataDAO.save(guest);
+			Assert.fail("This test should have failed");
+		}catch(RuntimeException e) {
+			Assert.assertNotNull(e);
+			String error="java.sql.SQLIntegrityConstraintViolationException: Column 'name' cannot be null";
+			Assert.assertEquals(error,e.getMessage());
+		}
+		bookingDataDAO.delete(bookingSaved.getId());
+	}
+	
+	@Test
+	public void testSaveBirthdateNull() {
+		BookingDataDTO bookingDataDTO=new BookingDataDTO(LocalDateTime.now(),LocalDateTime.of(2023,12,12,10,30),PaymentMethodDTO.CASH,new BigDecimal("6000"));
+		BookingDataDTO bookingSaved = bookingDataDAO.save(bookingDataDTO);
+		
+		GuestDataDTO guest=new GuestDataDTO("Juliana","Salinas",null,NationalityDTO.ARGENTIN,"1234567",bookingSaved.getId());
+		guestDataDAO.save(guest);
+		
+		guestDataDAO.delete(guest.getId());	
+		bookingDataDAO.delete(bookingSaved.getId());
+	}
+	
+	@Test
+	public void testSaveNationalityNull() {
+		BookingDataDTO bookingDataDTO=new BookingDataDTO(LocalDateTime.now(),LocalDateTime.of(2023,12,12,10,30),PaymentMethodDTO.CASH,new BigDecimal("6000"));
+		BookingDataDTO bookingSaved = bookingDataDAO.save(bookingDataDTO);
+		
+		GuestDataDTO guest=new GuestDataDTO("Juliana","Salinas",LocalDateTime.of(1990,12,12,10,30),null,"1234567",bookingSaved.getId());
+		guestDataDAO.save(guest);
```

Comment:
Unverified return, please do a select and verify the Nationality is actually null.